app/services/saml_request_validator.rb

```diff
@@ -5,17 +5,18 @@ class SamlRequestValidator
   validate :authorized_authn_context
   validate :authorized_email_nameid_format
 
-  def call(service_provider:, authn_context:, nameid_format:)
+  def call(service_provider:, authn_context:, nameid_format:, authn_context_comparison: 'exact')
```

Comment:
the way this is called, there will always be a value for `authn_context_comparison` but it might be nil, so this default needs to go inside as like `authn_context_comparison || 'exact'` or something to take effect

Reply:
I did it that way since the SAML spec says:
> If Comparison is set to "exact" or omitted, then the resulting authentication context in the authentication
statement MUST be the exact match of at least one of the authentication contexts specified.

It seemed easier to just default to a value of `exact` so we don't need to check presence and the string contents. Does that make sense?

Reply:
No I'm saying that...

```
authn_context_comparison = nil

SamlRequest.new.call(authn_context_comparison: authn_context_comparison)
```

Inside the method, the value will be `nil` instead of exact, unless you do something with `||=`

Reply:
If we're allowing the value to be omitted, then we should just remove the default argument

Reply:
Oh sorry, I see now (it's been a long Monday morning). I'll fix it, thanks for catching that!

Reply:
FYI - I left a default value of `nil` so that I wouldn't have to rewrite _all_ of the unit tests for the service object. I'll note it in the refactoring ticket.